src/jmath/Vec.ts

```diff
@@ -75,8 +75,11 @@ export function dotProduct(p1: Vec2, p2: Vec2): number {
   const angle = clockwiseAngle(getAngleBetweenVec2s(origin, p1), getAngleBetweenVec2s(origin, p2));
   return magnitude(p1) * magnitude(p2) * Math.cos(angle);
 }
+export function sqrMagnitude(p: Vec2): number {
+  return p.y * p.y + p.x * p.x;
+}
```

Comment:
I'm confused about this function and it's purpose.  Some comments here would be helpful for me.

Reply:
sqrMagnitude is magnitude without the sqrt() function. It's primarily used for optimization, as the sqrt() function is very demanding computationally, compared to the basic operators.

If you want to compare the lengths of two vectors, it's faster to compare sqrMagnitudes. If you want to check whether something is within a set distance you're supposed to compare (sqrMagnitude to distance * distance) instead of (magnitude to distance).

Modern computers are good enough that it's not a big deal 99.99% of the time, so usually I don't bother with it. Magnitude is usually a little more readable and nicer to work with. In the case of runForceMove however, sqrMagnitude might be a noticeable optimization, since it could be called 100 times each on 100 different units.

If anything needs the optimization, it's runForceMove 🤣